compiler/resolution/functionResolution.cpp

```diff
@@ -1664,27 +1825,55 @@ handleSymExprInExpandVarArgs(FnSymbol*  workingFn,
         newFormal->cname        = astr("_e", istr(i), "_", formal->cname);
 
         tupleCall->insertAtTail(new SymExpr(newFormal));
-
         formal->defPoint->insertBefore(newArgDef);
+        varargFormals[i] = newFormal;
       }
 
+      bool needTupleInBody;
+
       // Replace mappings to the old formal with mappings to the new variable.
       if (workingFn->hasFlag(FLAG_PARTIAL_COPY)) {
+        bool gotFormal = false; // for assertion only
```

Comment:
I recognize that this is a legal for statement but I think it is a poor decision to fold the loop update expression in to the loop test.  It's uncommon and easy to overlook.